database/src/main/postgres/runs/_validate_partitioning.sql

```diff
@@ -0,0 +1,130 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+CREATE OR REPLACE FUNCTION runs._validate_partitioning(
+    IN i_partitioning JSONB,
+    IN i_is_pattern BOOL = false
+) RETURNS SETOF TEXT AS
+$$
+-------------------------------------------------------------------------------
+--
+-- Function: runs._validate_partitioning(2)
+--      Validates the input partitioning and returns a set of individual error messages, one per an issue.
+--      The validation performs:
+--          1) Correct structure of the input JSONB object
+--          2) The list of keys in 'keys' is unique and doesn't have NULLs
+--          3) The keys in 'keys' and in 'keysToValuesMap' correspond to each other
+--          4) (if i_is_pattern = false) The values in 'keysToValuesMap' are non-empty/non-null
+--
+-- Parameters:
+--      i_partitioning      - partitioning to validate, a valid example:
+-- 	                          {
+--                              "keys": ["one", "two", "three"],
+--                              "version": 1,
+-- 	                            "keysToValuesMap": {
+--                                  "one": "DatasetA",
+--                                  "two": "Version1",
+-- 	                          	    "three": "2022-12-20"
+--                               }
+--                            }
+--      i_is_pattern        - flag signaling whether the partitioning is not a real one, but just a pattern.
+--                            If the input partitioning is just a pattern, then values can have NULL values and
+--                            such check would be skipped.
+--
+-- Returns:
+--      Set of issues described as individual textual sentences detailing each problem.
+--
+-------------------------------------------------------------------------------
+DECLARE
+    _mandatory_fields_in_input CONSTANT TEXT[] := ARRAY['keys', 'version', 'keysToValuesMap'];
+    _all_fields_in_input TEXT[];
+
+    _is_input_properly_structured BOOL;
+
+    _partitioning_keys_all_cnt INTEGER;
+    _partitioning_keys_uniq_and_not_null TEXT[];
+    _partitioning_keys_uniq_and_not_null_cnt INTEGER;
+
+    _partitioning_keys_from_values_map TEXT[];
+
+    _partitioning_null_values_cnt INTEGER;
+
+BEGIN
+    -- Checking whether the input has correct structure.
+    SELECT ARRAY(SELECT jsonb_object_keys(i_partitioning))
+    INTO _all_fields_in_input;
+
+    SELECT i_partitioning ?& _mandatory_fields_in_input
+    INTO _is_input_properly_structured;
+    IF NOT _is_input_properly_structured THEN
```

Comment:
It can of course. But I didn't want to do it, I think my code is much more readable because it assigns a meaning to some condition. One extra layer doesn't hurt in this case, on the other hand I think it's beneficial